Agent/Analytics/NRMAAnalytics.mm

```diff
@@ -268,7 +268,24 @@ - (BOOL) addInteractionEvent:(NSString*)name
         
         return [_eventManager addEvent:[event autorelease]];
     } else {
-        return _analyticsController->addInteractionEvent([name UTF8String], duration_secs, [self checkOfflineStatus], [self checkBackgroundStatus]);
+#if !TARGET_OS_WATCH
+        auto event = _analyticsController->addInteractionEvent([name UTF8String], duration_secs, [self checkBackgroundStatus]);
+        if([self checkOfflineStatus]){
+            event->addAttribute(kNRMA_Attrib_offline.UTF8String, @YES.boolValue);
+        }
+        
+        return _analyticsController->addEvent(event);
+
+#else
+        __block auto event = _analyticsController->addInteractionEvent([name UTF8String], duration_secs, [self checkBackgroundStatus]);
+        
+        [self checkOfflineStatus:^(BOOL isOffline) {
+            if(isOffline){
+                event->addAttribute(kNRMA_Attrib_offline.UTF8String, @YES.boolValue);
+            }
+        }];
+        return _analyticsController->addEvent(event);
```

Comment:
I think you want to add the event inside your completion handler. Otherwise I'm not sure the offline attribute will consistently be added.

Reply:
That is ideal, issue with that is that the function returns a bool indicating whether the event was added successfully. Do you know of a way we could maintain that? In the code we don't use that return value it's only used for unit testing.

Reply:
If the agent itself doesn't use it, then we might not needed it. But if you want to continue with it, have your completion handler take a bool indicating that.

Reply:
The return value is used and is important for custom events. I'm still not sure what the code would look like to have the return in the completion handler.